docs/reference/ingest/ingest-node.asciidoc

```diff
@@ -577,6 +577,554 @@ value of `service` to the value of the field `code`:
+--------------------------------------------------
+{
+  "_index": "test",
+  "_type": "_doc",
+  "_id": "1",
+  "_version": -3,
+  "result": "noop",
+  "_shards": {
+    "total": 0,
+    "successful": 0,
+    "failed": 0
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE
+
+
+[[ingest-conditional-nullcheck]]
+=== Handling Nested Fields in Conditionals
+
+Source documents often contain nested fields. Care should be taken
+to avoid NullPointerExceptions if the parent object does not exist
+in the document. For example `ctx.a.b.c` can throw an NullPointerExceptions
+if the source document does not have top level `a` object, or a second
+level `b` object.
+
+To help protect against NullPointerExceptions, null safe operations should be used.
+Fortunately Painless makes {painless}/painless-operators-reference.html#null-safe-operator[null safe]
+operations easy with the `?.` operator.
+
+[source,js]
+--------------------------------------------------
+PUT _ingest/pipeline/drop_guests_network
+{
+  "processors": [
+    {
+      "drop": {
+        "if": "ctx.network?.name == 'Guest'"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+The following document will get <<drop-processor,dropped>> correctly:
+
+[source,js]
+--------------------------------------------------
+POST test/_doc/1?pipeline=drop_guests_network
+{
+  "network": {
+    "name": "Guest"
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+////
+Hidden example assertion:
+[source,js]
+--------------------------------------------------
+GET test/_doc/1
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+// TEST[catch:missing]
+
+[source,js]
+--------------------------------------------------
+{
+  "_index": "test",
+  "_type": "_doc",
+  "_id": "1",
+  "found": false
+}
+--------------------------------------------------
+// TESTRESPONSE
+////
+
+Thanks to the `?.` operator the following document will not throw an error.
+If the pipeline used a `.` the following document would throw a NullPointerException
+since the `network` object is not part of the source document.
+
+[source,js]
+--------------------------------------------------
+POST test/_doc/2?pipeline=drop_guests_network
+{
+  "foo" : "bar"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+////
+Hidden example assertion:
+[source,js]
+--------------------------------------------------
+GET test/_doc/2
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+[source,js]
+--------------------------------------------------
+{
+  "_index": "test",
+  "_type": "_doc",
+  "_id": "2",
+  "_version": 1,
+  "found": true,
+  "_source": {
+    "foo": "bar"
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE
+////
+
+The source document can also use dot delimited fields to represent nested fields.
```

Comment:
I'm not so sure about this section. It's a nice explanation of the dot expander processor, but it seems to get a little off-topic? (it just seems to be a very specific example and I don't see how it's any different from explaining that you could use another script processor before your `if` to do a transformation on your doc before evaluation the conditional)

Reply:
The section is titled `Handling Nested Fields in Conditionals` and wanted to include this since it is (imo) not obvious.  Admittedly this ended up more of handling null references section and it is abit out of place. 

I don't have strong opinions of keeping it or tossing it, but lets see @acchen97 says about it's usefulness. If input documents are rarely in dot syntax then we should toss this, if it is common we can discuss how to integrate it better.